File: Python/Pac-Man/ECS/components/sprites.py
from ECS.ecs import *
from pygame import Color

class CircleSprite(Component):
	def __init__(self, radius: float, offset: Vector2 = Vector2(0, 0), color: Color = Color(0, 0, 0)):
		self.radius = radius
		self.offset = offset
		self.color = color

	def draw(self, surface: Surface):
		pygame.draw.circle(surface, self.color, tuple(self.owner.pos + self.offset), self.radius)

File: JS/Pong/ECS/components/sprites.js
import {Entity, Component} from "../ecs.js"

export class CircleSprite extends Component {
	constructor(radius) {
		super();
		this.radius = radius;
	}

	draw() {
		circle(this.owner.x, this.owner.y, this.radius * 2)
	}
}

export class RectSprite extends Component {
	constructor(w, h) {
		super();
		this.w = w;
		this.h = h;
	}

	draw() {
		rect(this.owner.x - this.w/2, this.owner.y - this.h/2, this.w, this.h);
	}
}

